devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -0,0 +1,142 @@
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap && descMap.clusterInt == 0x0006) {
+			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+				if (device.currentValue("level") == 0) {
+					sendEvent(name: "level", value: STEP)
+				}
+				sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		} else if (descMap && descMap.clusterInt == 0x0008) {
+			def currentLevel = device.currentValue("level") as Integer
+			if (descMap.commandInt == 0x02) {
+				def value = Math.min(currentLevel + STEP, 100)
+				log.debug "move to ${descMap.data}"
+				if (descMap.data[0] == "00") {
+					log.debug "move up"
+					sendEvent(name: "switch", value: "on")
+					sendEvent(name: "level", value: value)
+				} else if (descMap.data[0] == "01") {
+					log.debug "move down"
+					value = Math.max(currentLevel - STEP, 0)
+					sendEvent(name: "level", value: value)
+					if (value == 0) {
+						sendEvent(name: "switch", value: "off")
+					}
+				}
+			} else if (descMap.commandInt == 0x01) {
+//				def step = descMap.data[0] == "00" ? 10 : -10
+				sendEvent(name: "level", value: descMap.data[0] == "00" ? 100 : STEP)
+				sendEvent(name: "switch", value: "on" )
+//				def events = []
+//                for (currentLevel..target).step(step).each{
+//					events << createEvent(name: "level", value: currentLevel + it)
+//				}
+				log.debug "step to ${descMap.data}"
+			} else if (descMap.commandInt == 0x03) {
+				log.debug "stop move"
+			}
+		} else if (descMap && descMap.clusterInt == 0x0005) {
+			if (descMap.commandInt == 0x05) {
+				sendEvent(name: "button", value: "pressed", isStateChange: true)
+			} else if (descMap.commandInt == 0x04) {
+				sendEvent(name: "button", value: "held", isStateChange: true)
+			}
+		} else {
+			log.warn "DID NOT PARSE MESSAGE for description : $description"
+			log.debug "${descMap}"
+		}
+	}
+}
+
+def off() {
+	sendEvent(name: "switch", value: "off")
+}
+
+def on() {
+	if (device.currentValue("level") == 0) {
+		sendEvent(name: "level", value: STEP)
+	}
+	sendEvent(name: "switch", value: "on")
+}
+
+def setLevel(value) {
+	if (value != 0) sendEvent(name: "switch", value: "on")
+	sendEvent(name: "level", value: value)
+}
+
+def installed() {
+	sendEvent(name: "switch", value: "off", isStateChange: false, displayed: false)
+	sendEvent(name: "level", value: 0, isStateChange: false, displayed: false)
+	sendEvent(name: "button", value: "pressed", isStateChange: false, displayed: false)
+}
+
+def configure() {
+	log.debug "Configuring Reporting, IAS CIE, and Bindings."
+	def cmds = []
+	return zigbee.onOffConfig() +
```

Comment:
I don't think these configs will do anything since the device doesn't have On/Off or Level cluster servers, just clients.